src/main/webapp/app/tasks/task-form/task-form.component.spec.ts

```diff
@@ -33,39 +33,39 @@ describe('TaskFormComponent', () => {
 
   it('should validate a task', () => {
     expect(component.taskForm.invalid).toBe(true);
-    component.taskForm.setValue({name: 'My task'});
+    component.taskForm.setValue({name: 'My task', duedate: '18 Mar 2021'});
     expect(component.taskForm.invalid).toBe(false);
   });
 
   it('should create a task', () => {
     // given
-    component.taskForm.setValue({name: 'My task'});
-    taskService.create.and.returnValue(of({id: 'id', name: 'My task'}));
+    component.taskForm.setValue({name: 'My task', duedate: '18 Mar 2021'});
```

Comment:
Your main logic is within the tested file. You should at least adapt the test value of component.taskForm.value to something that is realistically returned from the browser, so that your formatting logic with moment.js is tested. Even better would be an additional test which tests only the correct date transformation with moment in the onSubmit method